### PR TITLE
Adding support for .Icon from Repo providers.

### PR DESCRIPTION
**Pull request by dhoehna**
## Summary of the pull request
Repository Providers can provide an icon via the `Icon` property on `IRepositoryProvider`.  However, the icon provided from the extension needs to be monochrome and colored for dark mode.

The extension repo will have a corresponding PR.

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

Movie!
UsingProviderIcon

### CODE REVIEW

File: tools/SetupFlow/DevHome.SetupFlow/Models/CloningInformation.cs [hunk 1]
```diff
@@ -83,33 +79,76 @@ public bool CloneToExistingDevDrive
...
         }
-        else
+
+        BitmapDecoder decoder = BitmapDecoder.CreateAsync(RepositoryProvider.Icon.OpenReadAsync().AsTask().Result).AsTask().Result;
+
+        // Get the pixel data as a byte array
+        PixelDataProvider pixelData = decoder.GetPixelDataAsync().AsTask().Result;
+        var pixels = pixelData.DetachPixelData();
+
+        // Assume that the icon is colored for dark mode.
```
Comment (EricJohnson327) [inline tools/SetupFlow/DevHome.SetupFlow/Models/CloningInformation.cs:116]: What happens if the icon is expected to have black inside of it?  Like if I have a smiley face icon with black eyes/smile, I wouldn't want the eyes/smile to change.

Comment (dhoehna) [reply tools/SetupFlow/DevHome.SetupFlow/Models/CloningInformation.cs:116]: This changes all black pixels to white and all white to black.  If the background is black, the face is white, and the eyes are black then, when in light mode the background would be white, face black, and eyes white.

The smile/eyes would still be visible, just different colors.  If we can get a icon like this I can test with it.

**Review by dkbennett — APPROVED** [review]

File: tools/SetupFlow/DevHome.SetupFlow/Models/CloningInformation.cs [hunk 2]
```diff
@@ -125,7 +164,7 @@ public void SetIcon(ElementTheme theme)
     public string RepositoryId => $"{RepositoryToClone.DisplayName ?? string.Empty}";
 
     /// <summary>
-    /// Gets the repository in a [organization]\[reponame] style
+    /// Gets the repository in a [oening account name]\[reponame] style
```
Comment (krschau) [inline tools/SetupFlow/DevHome.SetupFlow/Models/CloningInformation.cs:167]: typo in "owning"

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/RepoConfigViewModel.cs [hunk 3]
```diff
@@ -17,6 +17,7 @@
 using DevHome.SetupFlow.Utilities;
 using Microsoft.Extensions.Hosting;
 using Microsoft.UI.Xaml;
+using WinUIEx;
```
Comment (krschau) [inline tools/SetupFlow/DevHome.SetupFlow/ViewModels/RepoConfigViewModel.cs:20]: I'm guessing this got added automatically and overlooked.

**Review by krschau — APPROVED** [review]
Just two nits.

File: tools/SetupFlow/DevHome.SetupFlow/Models/CloningInformation.cs (hunk 2 above)
Comment (dhoehna) [reply tools/SetupFlow/DevHome.SetupFlow/Models/CloningInformation.cs:167]: Changed.

File: tools/SetupFlow/DevHome.SetupFlow/ViewModels/RepoConfigViewModel.cs (hunk 3 above)
Comment (dhoehna) [reply tools/SetupFlow/DevHome.SetupFlow/ViewModels/RepoConfigViewModel.cs:20]: Correct.  Thank you for catching this.

**Pull request merged**